demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1271 @@
+
+/**
+ * @brief Unsubscribe from the CreateCertificateFromCsr accepted and rejected topics.
+ */
+static bool unsubscribeFromCsrResponseTopics( void );
+
+/**
+ * @brief Subscribe to the RegisterThing accepted and rejected topics.
+ */
+static bool subscribeToRegisterThingResponseTopics( void );
+
+/**
+ * @brief Unsubscribe from the RegisterThing accepted and rejected topics.
+ */
+static bool unsubscribeFromRegisterThingResponseTopics( void );
+
+/**
+ * @brief Save the certificate to the path specified by #PROVISIONING_CERT_PATH.
+ */
+static bool saveCertificate( const char * certificate,
+                             size_t certificateLength );
+/*-----------------------------------------------------------*/
+
+static void provisioningPublishCallback( MQTTPublishInfo_t * pPublishInfo,
+                                         uint16_t packetIdentifier )
+{
+    FleetProvisioningStatus_t status;
+    FleetProvisioningTopic_t api;
+
+    /* Silence compiler warnings about unused variables. */
+    ( void ) packetIdentifier;
+
+    status = FleetProvisioning_MatchTopic( pPublishInfo->pTopicName,
+                                           pPublishInfo->topicNameLength, &api );
+
+    if( status != FleetProvisioningSuccess )
+    {
+        LogError( ( "Unexpected publish message received. Topic: %.*s.",
+                    ( int ) pPublishInfo->topicNameLength,
+                    ( const char * ) pPublishInfo->pTopicName ) );
+    }
+    else
+    {
+        if( api == FleetProvCborCreateCertFromCsrAccepted )
+        {
+            LogInfo( ( "Fleet Provisioning CreateCertificateFromCsr accepted." ) );
+
+            responseStatus = ResponseAccepted;
+
+            ( void ) memcpy( ( void * ) payloadBuffer,
+                             ( const void * ) pPublishInfo->pPayload,
+                             ( size_t ) pPublishInfo->payloadLength );
+
+            payloadLength = pPublishInfo->payloadLength;
+        }
+        else if( api == FleetProvCborCreateCertFromCsrRejected )
+        {
+            LogError( ( "Fleet Provisioning CreateCertificateFromCsr rejected." ) );
+
+            responseStatus = ResponseRejected;
+        }
+        else if( api == FleetProvCborRegisterThingAccepted )
+        {
+            LogInfo( ( "Fleet Provisioning RegisterThing accepted." ) );
+
+            responseStatus = ResponseAccepted;
+
+            ( void ) memcpy( ( void * ) payloadBuffer,
+                             ( const void * ) pPublishInfo->pPayload,
+                             ( size_t ) pPublishInfo->payloadLength );
+
+            payloadLength = pPublishInfo->payloadLength;
+        }
+        else if( api == FleetProvCborCreateCertFromCsrRejected )
+        {
+            LogError( ( "Fleet Provisioning RegisterThing rejected." ) );
+
+            responseStatus = ResponseRejected;
+        }
+        else
+        {
+            LogError( ( "Unexpected Fleet Provisioning message. Topic: %.*s.",
```

Comment:
This path should not be reached, should it? 
```suggestion
            LogError( ( "Failed to parse incoming MQTT Publish topic from AWS IoT Core. Topic: %.*s.",
```

Reply:
This should not be reached. However, this case is not a parse failure. It is when we get a message that is from Fleet Provisioning, but it is not on a Fleet Provisioning topic we were expecting a response on